src/storage-folder.coffee

```diff
@@ -0,0 +1,27 @@
+path = require "path"
+fs = require "fs-plus"
+
+module.exports =
+class StorageFolder
+  constructor: (containingPath) ->
+    @path = path.join(containingPath, "storage")
+
+  store: (name, object) ->
+    fs.writeFileSync(@pathForKey(name), JSON.stringify(object), 'utf8')
```

Comment:
I think the `Atom.saveSync` code was very very old, and as we've learned since then, the fs is far from reliable writing stuff, https://github.com/atom/atom/blob/c665747879161991f413db2a967527c8abdae1c0/src/pane.coffee#L666-L682